### ISSUE TITLE
generate_test_main.go: no such file or directory

### ISSUE THREAD

**Issue by jakevoytko** [bazelbuild/rules_go#39]
Unable to get any git tests to build at HEAD, even in the following simple project with 4 files:

**WORKSPACE**
```
git_repository(
    name = "io_bazel_rules_go",
    remote = "https://github.com/bazelbuild/rules_go.git",
    tag = "0.0.2",
)
load("@io_bazel_rules_go//go:def.bzl", "go_repositories")

go_repositories()
```

**BUILD**
```
load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")

go_prefix("github.com/jakevoytko/bazel-test")

go_library(name="go_default_library",srcs=["test.go"])

go_test(name="test", srcs=["go_test.go"], deps=[":go_default_library"])
```

**test.go**
```
package bazel

func UnderTest(i int) int {
	return i + 2
}
```

**go_test.go**
```
package bazel

import (
	"fmt"
	"testing"
)

func TestTheThing(t *testing.T) {
	fmt.Println("hello")
}
```

command output:
```
Jacobs-MacBook-Pro:bazel jake$ bazel test :test
.
INFO: Found 1 test target...
ERROR: /private/var/tmp/_bazel_jake/5a307faf9a56c70a9ba44e4f95d550a5/external/io_bazel_rules_go/go/tools/BUILD:6:1: error executing shell command: 'rm -rf bazel-out/host/io_bazel_rules_go/go/tools/generate_test_main.a.dir && mkdir -p bazel-out/host/io_bazel_rules_go/go/tools/generate_test_main.a.dir && mkdir -p bazel-out/host/io_bazel_rules_go...' failed: bash failed: error executing command /bin/bash -c ... (remaining 1 argument(s) skipped): com.google.devtools.build.lib.shell.BadExitStatusException: Process exited with status 1.
open github.com/bazelbuild/rules_go/../io_bazel_rules_go/go/tools/generate_test_main.go: open github.com/bazelbuild/rules_go/../io_bazel_rules_go/go/tools/generate_test_main.go: no such file or directory
Target //:test failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 14.788s, Critical Path: 0.04s
```

bazel version:
```
Jacobs-MacBook-Pro:bazel jake$ bazel version
Build label: 0.3.0-2016-06-18 (@ff1332d)
Build target: bazel-out/local-fastbuild/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Sun Jun 19 00:09:35 2016 (1466294975)
Build timestamp: 1466294975
Build timestamp as int: 1466294975
```

**Comment by yugui** [comment]
It is a bug fixed in v0.0.3.  You can built the test program if you update `rules_go` to 0.0.3.
However I noticed there is another bug that the test driver tries to invoke `/bin/test` instead of your test program and it exits with 1.

I'll fix this bug later, but you can avoid this bug by defining your test program in a subpackage as a workaround.

**Comment by yugui** [comment]
It looked to be a bug of Bazel rather than a bug of `rules_go`.
I have reported the bug at bazelbulid/bazel#1443
